test/org/zaproxy/zap/extension/pscanrules/PasswordAutocompleteScannerUnitTest.java

```diff
@@ -0,0 +1,260 @@
+	}
+
+	@Test
+	public void shouldRaiseAlertIfAutocompleteIsInWrongTag() throws HttpMalformedHeaderException {
+		String body =
+				"<form>" +
+				"<input name='usr' type='text' autocomplete='off'>" +
+				"<input name='pw' type='password'>" +
+				"</form>";
+		HttpMessage msg = createHttpMessage(body);
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+		assertThat(alertsRaised.size(), equalTo(1));
+		assertThat(alertsRaised.get(0).getParam(), equalTo("pw"));
+		assertThat(alertsRaised.get(0).getEvidence(), equalTo("<input name='pw' type='password'>"));
+	}
+
+	@Test
+	public void shouldRaiseAlertIfOneAutocompleteOfFormTagInTwoForms() throws HttpMalformedHeaderException {
+		String body =
+				"<form action='login1' autocomplete='off'>" +
+				"<input name='pw1' id='aId1' type='password'>" +
+				"</form>" +
+				"<form action='login2'>" +
+				"<input name='pw2' id='aId2' type='password'>" +
+				"</form>";
+		HttpMessage msg = createHttpMessage(body);
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+		assertThat(alertsRaised.size(), equalTo(1));
+		assertThat(alertsRaised.get(0).getParam(), equalTo("aId2"));
+		assertThat(alertsRaised.get(0).getEvidence(), equalTo("<input name='pw2' id='aId2' type='password'>"));
+	}
+
+	@Test
+	public void shouldRaiseAlertIfNoAutocompleteInTwoForms() throws HttpMalformedHeaderException {
+		String body =
+				"<form action='login1'>" +
+				"<input name='pw1' type='password'>" +
+				"</form>" +
+				"<form action='login2'>" +
+				"<input name='pw2' type='password'>" +
+				"</form>";
+		HttpMessage msg = createHttpMessage(body);
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+		String [][] checkValues = new String [][] {
+				{"pw1", "<input name='pw1' type='password'>"},
+				{"pw2", "<input name='pw2' type='password'>"}
+		};
+		assertThat(alertsRaised.size(), equalTo(2));
+		for (int i = 0; i < alertsRaised.size(); i++) {
+			assertThat(alertsRaised.get(i).getParam(), equalTo(checkValues[i][0]));
+			assertThat(alertsRaised.get(i).getEvidence(), equalTo(checkValues[i][1]));
+			validateAlert(alertsRaised.get(i));
+		}
+	}
+
+	@Test
+	public void shouldRaiseAlertIfOneAutocompleteOfPasswordFieldInTwoForms() throws HttpMalformedHeaderException {
+		String body =
+				"<form action='login1'>" +
+				"<input name='pw1' type='password' autocomplete='off'>" +
+				"</form>" +
+				"<form action='login2'>" +
+				"<input name='pw2' type='password'>" +
+				"</form>";
+		HttpMessage msg = createHttpMessage(body);
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+		assertThat(alertsRaised.size(), equalTo(1));
+		assertThat(alertsRaised.get(0).getParam(), equalTo("pw2"));
+		assertThat(alertsRaised.get(0).getEvidence(), equalTo("<input name='pw2' type='password'>"));
+	}
+
+
+	@Test
+	// No real test, just in case that a code coverage tool will be introduced later on
+	public void TestOfPassesIfAwsHostnameTest() throws HttpMalformedHeaderException {
```

Comment:
Sorry, I'm not following. It seems like the two pulls are getting crossed up. How does this aws  piece first into password auto complete?